TFNetwork.py

```diff
@@ -1964,14 +1964,23 @@ def __init__(self, filename, saveable_params, params_prefix="", load_if_prefix="
     self.params_prefix = params_prefix
     self.load_if_prefix = load_if_prefix
     self.saveable_params = []
+    self.param_mapping = {}
+    layer_mapping = layer_mapping or {}
     for param in saveable_params:
       custom_post_init = getattr(param, "custom_post_init", None)
       if custom_post_init:
         print("Not loading pre-initialized variables %s" % param, file=log.v2)
         continue
-      if load_if_prefix and self._get_param_name(param, assert_load_if_prefix_match=False) is None:
+      param_name = self._get_param_name(param, assert_load_if_prefix_match=False)
+      if load_if_prefix and param_name is None:
         continue
       self.saveable_params.append(param)
+      param_layer_name = "/".join(param_name.split("/")[:-1])
+      if param_layer_name in layer_mapping:
+        self.param_mapping[param_name] = layer_mapping[param_layer_name] + "/" + param_name.split("/")[-1]
+      else:
+        self.param_mapping[param_name] = param_name
```

Comment:
This code is too complicated and too specific and also prone to errors. It should be more generic.

Reply:
Line 1980 could be replaced with `self.param_mapping[param_name] = param_name.replace(param_layer_name, layer_mapping[param_layer_name])`. This should make clearer that we're modifying the name of the layer.
Do you agree?

Reply:
No, what I mean is esp `param_layer_name = "/".join(param_name.split("/")[:-1])`. This is too arbitrary. This is also wrong in some cases. Also, this is too complicated.
I am not sure if I would introduce this layer mapping at all. Maybe param mapping, which is simpler.